### ISSUE TITLE
An example on recon osem parameters num_block and ind_block appreciated

### ISSUE THREAD

**Issue by katerinachytra** [tomopy/tomopy#539]
Hi all, please, could you provide an example on how to set the parameters num_block and ind_block while using recon with algorithm osem? I am trying to reconstruct 3D data and as I understood, the num_block is the number of subsets. However, if I set this parameter different from 1, the result is a blank image. 
Thank you in advance. Katerina

**Comment by erangraz** [comment]
I have the same problem. When using OSEM algorithm with num_block or num_iter more than one, the output is filled with NAN. I am wondering if someone looks into this problem.

**Comment by dgursoy** [comment]
How many projection angles do you have? This number should be divisible to num_block.

**Comment by carterbox** [comment]
As discussed privately, there is an annoying division that is currently unchecked.

https://github.com/tomopy/tomopy/blob/d04e73c362189fe4802b644332a44029172c447d/source/libtomo/recon/osem.c#L164-L172

@dgursoy, There are two mitigation options that are possible. Which one is better?

```c
// Proposed Mitigation 1
                        if(sum_dist2 != 0.0f)
                        {
                            ind_data = d + p * dx + s * dt * dx;
                            upd      = data[ind_data] / (simdata[ind_data] + 1e-6F);
                            for(n = 0; n < csize - 1; n++)
                            {
                                update[indi[n]] += upd * dist[n];
                            }
                        }
```

```c
// Proposed Mitigation 2
                        if(sum_dist2 != 0.0f && sumdata[ind_data] != 0.0f)
                        {
                            ind_data = d + p * dx + s * dt * dx;
                            upd      = data[ind_data] / simdata[ind_data];
                            for(n = 0; n < csize - 1; n++)
                            {
                                update[indi[n]] += upd * dist[n];
                            }
                        }
```

**Comment by carterbox** [comment]
UPDATE: The fixes that I provided in #545 do not actually address the instability encountered with the ordered subset methods. I believe we cannot fix the instability without changing the algorithm, so the actual solution to this problem is to use `num_block=1` if instability is observed.
